src/js/game/LevelMVC/LevelModel.js

```diff
@@ -258,6 +258,42 @@ module.exports = class LevelModel {
     return this.controller.score;
   }
 
+  shouldRide(direction) {
+    let player = this.player;
+    let frontPosition = this.getNextRailPosition(player);
+    let frontBlock = this.actionPlane.getBlockAt(frontPosition);
+    return this.isNextRailValid(frontBlock, direction);
+  }
+
+  isNextRailValid(block, direction) {
+    if (!block) {
+      return;
+    }
+    return FacingDirection.opposite(block.connectionA) === direction ||
+      FacingDirection.opposite(block.connectionB) === direction ||
+      block.connectionA === direction ||
+      block.connectionB === direction;
+  }
+
+  getNextRailPosition(entity = this.player) {
+    let offset = [0,0];
+    switch (entity.movementState) {
+      case 0:
+        offset[1] = -1;
+        break;
+      case 1:
+        offset[0] = 1;
+        break;
+      case 2:
+        offset[1] = 1;
+        break;
+      case 3:
+        offset[0] = -1;
+        break;
+    }
+    return Position.add(entity.position, offset);
+  }
```

Comment:
(replying to https://github.com/code-dot-org/craft/pull/332#discussion_r147794109 )

I'd still rather avoid duplicating so much code. How about just:
```javascript
const offset = FacingDirection.directionToOffset(entity.movementState) || [0, 0];
```

Reply:
That works too. 🙆‍♂️